### ISSUE TITLE
Building for target  armv7-unknown-linux-gnueabihf error

### ISSUE THREAD

**Issue by heibor** [khvzak/mlua#14]
It seems that the mlua could not build for target  armv7-unknown-linux-gnueabihf. When I executed:

```
cargo build --target armv7-unknown-linux-gnueabihf
```

I got these error message:

```
 running: "arm-linux-gnueabihf-ar" "crs" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/liblua5.3.a" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lapi.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lauxlib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lbaselib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lcode.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/ldblib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/ldebug.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/ldo.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/ldump.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lfunc.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lgc.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/linit.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/liolib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/llex.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lmathlib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lmem.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/loadlib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lobject.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lopcodes.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/loslib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lparser.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lstate.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lstring.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lstrlib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/ltable.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/ltablib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/ltm.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lundump.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lvm.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lzio.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lbitlib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lcorolib.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lctype.o" "/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib/lutf8lib.o"
  exit code: 0
  cargo:rustc-link-search=native=/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib
  cargo:rustc-link-lib=static=lua5.3
  cargo:include=/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/include
  cargo:lib=/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/lua-build/lib
  OPT_LEVEL = Some("0")
  TARGET = Some("armv7-unknown-linux-gnueabihf")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_armv7-unknown-linux-gnueabihf = None
  CC_armv7_unknown_linux_gnueabihf = None
  TARGET_CC = None
  CC = None
  CROSS_COMPILE = None
  CFLAGS_armv7-unknown-linux-gnueabihf = None
  CFLAGS_armv7_unknown_linux_gnueabihf = None
  TARGET_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  CARGO_CFG_TARGET_FEATURE = None

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "The command \"/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/glue\" \"/home/zhen/works/rust/t01/target/armv7-unknown-linux-gnueabihf/debug/build/mlua-1e626287640eeec6/out/glue.rs\" did not run successfully." }', mlua/build/main.rs:91:10
  stack backtrace:
     0: backtrace::backtrace::libunwind::trace
               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
     1: backtrace::backtrace::trace_unsynchronized
               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
     2: std::sys_common::backtrace::_print_fmt
               at src/libstd/sys_common/backtrace.rs:78
     3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
               at src/libstd/sys_common/backtrace.rs:59
     4: core::fmt::write
               at src/libcore/fmt/mod.rs:1076
     5: std::io::Write::write_fmt
               at src/libstd/io/mod.rs:1537
     6: std::sys_common::backtrace::_print
               at src/libstd/sys_common/backtrace.rs:62
     7: std::sys_common::backtrace::print
               at src/libstd/sys_common/backtrace.rs:49
     8: std::panicking::default_hook::{{closure}}
               at src/libstd/panicking.rs:198
     9: std::panicking::default_hook
               at src/libstd/panicking.rs:217
    10: std::panicking::rust_panic_with_hook
               at src/libstd/panicking.rs:526
    11: rust_begin_unwind
               at src/libstd/panicking.rs:437
    12: core::panicking::panic_fmt
               at src/libcore/panicking.rs:85
    13: core::option::expect_none_failed
               at src/libcore/option.rs:1269
    14: core::result::Result<T,E>::unwrap
               at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9/src/libcore/result.rs:1005
    15: build_script_main::build_glue
               at mlua/build/main.rs:88
    16: build_script_main::main
               at mlua/build/main.rs:135
    17: std::rt::lang_start::{{closure}}
               at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9/src/libstd/rt.rs:67
    18: std::rt::lang_start_internal::{{closure}}
               at src/libstd/rt.rs:52
    19: std::panicking::try::do_call
               at src/libstd/panicking.rs:348
    20: std::panicking::try
               at src/libstd/panicking.rs:325
    21: std::panic::catch_unwind
               at src/libstd/panic.rs:394
    22: std::rt::lang_start_internal
               at src/libstd/rt.rs:51
    23: std::rt::lang_start
               at /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9/src/libstd/rt.rs:67
    24: main
    25: __libc_start_main
    26: _start
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by khvzak** [comment]
At the moment cross compilation is not supported (and I'm working on it!), because during the mlua compilation, the small program called `glue` is executed to determine system constants and generate `glue.rs` file afterwards. This is the main blocker, and seems one of the solution could be pre-defining the constants for each pair lua,os.

**Comment by wez** [comment]
This also blocks building for "Apple Silicon" from an Intel based mac (https://github.com/wez/wezterm/pull/426 is my experiment to make that work).  Building for Intel from Apple Silicon on my local machine works because that system can run both architectures, but github actions doesn't have a native M1 target at the moment.

Do you have a general idea in mind for how this should be resolved?

**Comment by khvzak** [comment]
I've released v0.5.1 that introduces cross-compilation support for the **vendored** mode, thanks to @wez.
